lapack/src/KokkosLapack_svd.hpp

```diff
@@ -0,0 +1,238 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+/// \file KokkosLapack_svd.hpp
+/// \brief Singular Value Decomposition (SVD)
+///
+/// This file provides KokkosLapack::svd. This function performs a
+/// local (no MPI) singular value decomposition of the input matrix A
+/// and returns the singular values and vectors dedending on input flags.
+
+#ifndef KOKKOSLAPACK_SVD_HPP_
+#define KOKKOSLAPACK_SVD_HPP_
+
+#include <type_traits>
+
+#include "KokkosLapack_svd_spec.hpp"
+#include "KokkosKernels_Error.hpp"
+
+namespace KokkosLapack {
+
+/// \brief Compute the Singular Value Decomposition of A = U*S*Vt
+///
+/// \tparam ExecutionSpace the space where the kernel will run.
+/// \tparam AMatrix (mxn) matrix as a rank-2 Kokkos::View.
+/// \tparam SVector min(m,n) vector as a rank-1 Kokkos::View
+/// \tparam UMatrix (mxm) matrix as a rank-2 Kokkos::View
+/// \tparam VMatrix () matrix as a rank-2 Kokkos::View
+///
+/// \param space [in] execution space instance used to specified how to execute
+///   the svd kernels.
+/// \param jobu [in] flag to control the computation of the left singular
+/// vectors when set to: 'A' all vectors are computed, 'S' the first min(m,n)
+/// singular vectors are computed, 'O' the first min(m,n) singular vectors are
+/// overwritten into A, 'N' no singular vectors are computed. \param jobvt [in]
+/// flag to control the computation of the right singular vectors when set to:
+/// 'A' all vectors are computed, 'S' the first min(m,n) singular vectors are
+/// computed, 'O' the first min(m,n) singular vectors are overwritten into A,
+/// 'N' no singular vectors are computed. \param jobvt [in] \param A [in] An
```

Comment:
Looks like the auto-formatter mangled the doxygen style argument list - not sure if doxygen will work with a ``\param`` starting in the middle of a line.